docs/ci/ci-code-coverage-reporting.md

```diff
@@ -0,0 +1,138 @@
+
+## Target Goals
+
+By implementing code coverage reporting:
+
+* Contributors should be able to _easily_ identify which new lines of code in their branch need more coverage.
+  * This means contributors should not need to dig through CI logging to determine code coverage.
+  * Ideally, a bot should leave a comment to summarize the code coverage for them OR the contributor should be able to find this information without having to search through text (e.g., logging).
+* Contributors should understand how well their branch code is covered relative to the current `main` branch code.
+
+## High-Level Functional Requirements
+
+* Code coverage should be reported in PR builds as a GitHub _bot_ comment
+* New code:
+  * Should show % coverage on new lines of code.
+  * Should show which new lines of code are uncovered (if any).
+* Existing code:
+  * Should show existing % coverage on all lines of code.
+* New code + existing code:
+  * Should show projected % coverage if the PR is to be merged (e.g., “overall code coverage 53% -> 62% after merge”).
+* The GitHub coverage reporting bot should be able to interpret at least one of these pytest coverage reporting formats:
+  * HTML
+  * XML
+  * JSON
+  * lcov
+  * annotate
+* The README should display a code coverage badge
+
+## Code Coverage Reporting Bot: Options
+
+### [Codecov](https://github.com/marketplace/codecov)
+
+* [Example Codecov job](https://github.com/ansible/vscode-ansible/runs/33887209843)
+* Pros:
+  * Free to use in open source projects.
+  * Codecov reports all the data on GitHub in a comment. [Example](https://docs.codecov.com/docs/pull-request-comments).
+* Cons / Callouts:
+  * Not applicable at this time, but if we want to use it in >1 private repo, we must purchase a customer plan.
+  * According to Codecov: "All subscribers of a Pull Request get an email when Codecov posts a comment (only when posting a new comment), and there's nothing we can do about it.” (Read more [here](https://docs.codecov.com/docs/pull-request-comments#filtering-emails).)
+
+### [Coveralls](https://github.com/marketplace/actions/coveralls-github-action)
+
+* [Example Coveralls job](https://github.com/gophercloud/gophercloud/actions/runs/12117814428/job/33858835625?pr=3244)
+* Pros:
+  * Free to use via [MIT license here](https://github.com/coverallsapp/github-action/blob/main/LICENSE.md).
+  * Appears to be free for open source repos, but like with Codecov, we should validate that the ToS doesn’t have fine print, etc.
+  * Can report all the data in a GitHub comment. [Example](https://github.com/coverallsapp/coveralls-node-demo/pull/19#issuecomment-1035344985).
+* Cons / Callouts:
+  * Code coverage visuals, etc. are all viewed through the Coveralls front end (coveralls.io), so if that service were to go down, then we’re technically at their mercy.
```

Comment:
If it's FOSS/MIT licensed, could we not stand up our own instance?

Reply:
True, but I'd prefer to avoid the overhead (is possible)